sycl/include/CL/sycl/INTEL/esimd/esimd_libcm.hpp

```diff
@@ -0,0 +1,311 @@
+LAMBDA_WRAPPER_TMPL(sycl::item<2 _COMMA_ true>, ITEM_OFFSET_2DIM)
+LAMBDA_WRAPPER_TMPL(sycl::item<1 _COMMA_ true>, ITEM_OFFSET_1DIM)
+LAMBDA_WRAPPER_TMPL(sycl::nd_item<3>, NDITEM_3DIM)
+LAMBDA_WRAPPER_TMPL(sycl::nd_item<2>, NDITEM_2DIM)
+LAMBDA_WRAPPER_TMPL(sycl::nd_item<1>, NDITEM_1DIM)
+
+template <class KernelType, class KernelArgType, typename IteratorType,
+          int __Dims__>
+class libCMBatch {
+private:
+  KernelType MKernel;
+  std::vector<IteratorType> argVector;
+  std::vector<uint32_t> spaceDim;
+  const std::vector<uint32_t> singleGrpDim = {1, 1, 1};
+  const uint32_t hwThreads = (uint32_t)std::thread::hardware_concurrency();
+
+  using IDBuilder = sycl::detail::Builder;
+
+public:
+  libCMBatch(KernelType Kernel) : MKernel(Kernel), spaceDim{1, 1, 1} {}
+
+  // ID
+  template <class ArgT = KernelArgType>
+  typename std::enable_if<std::is_same<ArgT, sycl::id<__Dims__>>::value>::type
+  runIterationSpace(const sycl::range<__Dims__> Range) {
+    sycl::detail::NDLoop<__Dims__>::iterate(
+        Range, [&](const sycl::id<__Dims__> &ID) { argVector.push_back(ID); });
+    for (int I = 0; I < __Dims__; ++I) {
+      spaceDim[I] = (uint32_t)Range[I];
+    }
+    run();
+  }
+
+  // Item w/o offset
+  template <class ArgT = KernelArgType>
+  typename std::enable_if<
+      std::is_same<ArgT, sycl::item<__Dims__, /*Offset=*/false>>::value>::type
+  runIterationSpace(const sycl::range<__Dims__> Range) {
+    sycl::detail::NDLoop<__Dims__>::iterate(
+        Range, [&](const sycl::id<__Dims__> ID) {
+          sycl::item<__Dims__, /*Offset=*/false> Item =
+              IDBuilder::createItem<__Dims__, false>(Range, ID);
+          argVector.push_back(Item);
+        });
+    for (int I = 0; I < __Dims__; ++I) {
+      spaceDim[I] = (uint32_t)Range[I];
+    }
+    run();
+  }
+
+  // Item w/ offset
+  template <class ArgT = KernelArgType>
+  typename std::enable_if<
+      std::is_same<ArgT, sycl::item<__Dims__, /*Offset=*/true>>::value>::type
+  runIterationSpace(const sycl::range<__Dims__> Range,
+                    const sycl::id<__Dims__> Offset) {
+    sycl::detail::NDLoop<__Dims__>::iterate(
+        Range, [&](const sycl::id<__Dims__> &ID) {
+          sycl::id<__Dims__> OffsetID = ID + Offset;
+          sycl::item<__Dims__, /*Offset=*/true> Item =
+              IDBuilder::createItem<__Dims__, true>(Range, OffsetID, Offset);
+          argVector.push_back(Item);
+        });
+    for (int I = 0; I < __Dims__; ++I) {
+      spaceDim[I] = (uint32_t)Range[I];
+    }
+    run();
+  }
+
+  // NDItem
+  template <class ArgT = KernelArgType>
+  typename std::enable_if<
+      std::is_same<ArgT, sycl::nd_item<__Dims__>>::value>::type
+  runIterationSpace(const sycl::id<__Dims__> &GroupID,
+                    const sycl::group<__Dims__> Group,
+                    const sycl::range<__Dims__> LocalSize,
+                    const sycl::range<__Dims__> GlobalSize,
+                    const sycl::id<__Dims__> GlobalOffset) {
```

Comment:
```suggestion
                    const sycl::group<__Dims__> &Group,
                    const sycl::range<__Dims__> &LocalSize,
                    const sycl::range<__Dims__> &GlobalSize,
                    const sycl::id<__Dims__> &GlobalOffset) {
```

Reply:
Could you please apply in all such cases?

Reply:
Okay. Will apply '&'.